Update var to let/const

adeobas opened the pull request:
<!-- 👀 Thanks for opening a PR! Read comments like this one to get your PR merged faster. -->
#### Summary
Updated page to replace "var" with "let" or "const" 
<!-- ✍️ In a sentence or two, describe your changes -->

#### Motivation
<!-- ❓ Why are you making this change? Help us understand how your changes help readers. -->
It is bad practice to use var. 

#### Supporting details
<!-- 🔗 Link to supporting information, such as bug trackers, source control, release notes, and vendor docs. -->

#### Related issues
<!-- 🔨 If applicable, use "Fixes #XYZ" -->

#### Metadata
<!-- ✅ Check a box if applicable, like this: [x]

This PR…
-->
- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

<!-- 👷‍♀️ After submitting, review the results of the "Checks" tab! -->

adeobas committed:
Update var to let/const

queengooborg commented on files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:
```diff
@@ -11,8 +11,6 @@ tags:
   - speech
   - synthesis
 ---
-{{DefaultAPISidebar("Web Speech API")}}
```
Accidental removal?

adeobas replied:
Thank you for your suggestions. I reviewed them and approved the suggestions. 
I didn't mean to remove. I submitted a separate PR request to add the sidebar. I'm assuming that the update var to let/const removed it?

queengooborg replied:
No worries, it happens!  You can add it back in by going to `Files changed` > `...` (top-right corner) > `Edit file`!  (Apparently I can't make a suggestion here, don't know why...)

queengooborg commented on files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:
```diff
@@ -58,18 +56,18 @@ Let's look at the JavaScript in a bit more detail.
-var SpeechRecognition = window.SpeechRecognition || webkitSpeechRecognition;
-var SpeechGrammarList = window.SpeechGrammarList || webkitSpeechGrammarList;
-var SpeechRecognitionEvent = window.SpeechRecognitionEvent || webkitSpeechRecognitionEvent;
+let SpeechRecognition = window.SpeechRecognition || webkitSpeechRecognition;
+let SpeechGrammarList = window.SpeechGrammarList || webkitSpeechGrammarList;
+let SpeechRecognitionEvent = window.SpeechRecognitionEvent || webkitSpeechRecognitionEvent;
```
```suggestion
const SpeechRecognition = window.SpeechRecognition || webkitSpeechRecognition;
const SpeechGrammarList = window.SpeechGrammarList || webkitSpeechGrammarList;
const SpeechRecognitionEvent = window.SpeechRecognitionEvent || webkitSpeechRecognitionEvent;
```

queengooborg commented on files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:
````diff
@@ -58,18 +56,18 @@ Let's look at the JavaScript in a bit more detail.
 ```js
-var colors = [ 'aqua' , 'azure' , 'beige', 'bisque', 'black', 'blue', 'brown', 'chocolate', 'coral' ... ];
-var grammar = '#JSGF V1.0; grammar colors; public <color> = ' + colors.join(' | ') + ' ;'
+let colors = [ 'aqua' , 'azure' , 'beige', 'bisque', 'black', 'blue', 'brown', 'chocolate', 'coral' ... ];
+let grammar = '#JSGF V1.0; grammar colors; public <color> = ' + colors.join(' | ') + ' ;'
````
```suggestion
const colors = [ 'aqua' , 'azure' , 'beige', 'bisque', 'black', 'blue', 'brown', 'chocolate', 'coral' ... ];
const grammar = '#JSGF V1.0; grammar colors; public <color> = ' + colors.join(' | ') + ' ;'
```

queengooborg commented on files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:
````diff
@@ -84,8 +82,8 @@ The grammar format used is [JSpeech Grammar Format](https://www.w3.org/TR/jsgf/)
 ```js
-var recognition = new SpeechRecognition();
-var speechRecognitionList = new SpeechGrammarList();
+let recognition = new SpeechRecognition();
+let speechRecognitionList = new SpeechGrammarList();
````
```suggestion
const recognition = new SpeechRecognition();
const speechRecognitionList = new SpeechGrammarList();
```

queengooborg commented on files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:
```diff
@@ -114,11 +112,11 @@ recognition.maxAlternatives = 1;
-var diagnostic = document.querySelector('.output');
-var bg = document.querySelector('html');
-var hints = document.querySelector('.hints');
+let diagnostic = document.querySelector('.output');
+let bg = document.querySelector('html');
+let hints = document.querySelector('.hints');
```
```suggestion
const diagnostic = document.querySelector('.output');
const bg = document.querySelector('html');
const hints = document.querySelector('.hints');
```

queengooborg commented on files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:
````diff
@@ -137,7 +135,7 @@ Once the speech recognition is started, there are many event handlers that can b
 ```js
 recognition.onresult = function(event) {
-  var color = event.results[0][0].transcript;
+  let color = event.results[0][0].transcript;
````
```suggestion
  const color = event.results[0][0].transcript;
```

queengooborg commented on files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:
````diff
@@ -230,18 +228,18 @@ Let's investigate the JavaScript that powers this app.
 
 ```js
-var synth = window.speechSynthesis;
+let synth = window.speechSynthesis;
````
```suggestion
const synth = window.speechSynthesis;
```

queengooborg commented on files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:
```diff
@@ -230,18 +228,18 @@ Let's investigate the JavaScript that powers this app.
-var inputForm = document.querySelector('form');
-var inputTxt = document.querySelector('.txt');
-var voiceSelect = document.querySelector('select');
+let inputForm = document.querySelector('form');
+let inputTxt = document.querySelector('.txt');
+let voiceSelect = document.querySelector('select');
```
```suggestion
const inputForm = document.querySelector('form');
const inputTxt = document.querySelector('.txt');
const voiceSelect = document.querySelector('select');
```

queengooborg commented on files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:
```diff
@@ -230,18 +228,18 @@ Let's investigate the JavaScript that powers this app.
-var pitchValue = document.querySelector('.pitch-value');
-var rate = document.querySelector('#rate');
-var rateValue = document.querySelector('.rate-value');
+let pitch = document.querySelector('#pitch');
+let pitchValue = document.querySelector('.pitch-value');
+let rate = document.querySelector('#rate');
+let rateValue = document.querySelector('.rate-value');
```
```suggestion
const pitch = document.querySelector('#pitch');
const pitchValue = document.querySelector('.pitch-value');
const rate = document.querySelector('#rate');
const rateValue = document.querySelector('.rate-value');
```

queengooborg commented on files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:
```diff
@@ -230,18 +228,18 @@ Let's investigate the JavaScript that powers this app.
+let rateValue = document.querySelector('.rate-value');
 
-var voices = [];
+let voices = [];
```
```suggestion
const voices = [];
```

queengooborg commented on files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:
```diff
@@ -255,7 +253,7 @@ function populateVoiceList() {
 
   for(i = 0; i < voices.length ; i++) {
-    var option = document.createElement('option');
+    let option = document.createElement('option');
```
```suggestion
    const option = document.createElement('option');
```

queengooborg commented on files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md:
```diff
@@ -290,8 +288,8 @@ Finally, we set the {{domxref("SpeechSynthesisUtterance.pitch")}} and {{domxref(
 
-  var utterThis = new SpeechSynthesisUtterance(inputTxt.value);
-  var selectedOption = voiceSelect.selectedOptions[0].getAttribute('data-name');
+  let utterThis = new SpeechSynthesisUtterance(inputTxt.value);
+  let selectedOption = voiceSelect.selectedOptions[0].getAttribute('data-name');
```
```suggestion
  const utterThis = new SpeechSynthesisUtterance(inputTxt.value);
  const selectedOption = voiceSelect.selectedOptions[0].getAttribute('data-name');
```

queengooborg reviewed (changes requested):
Thank you for your PR!  I see a lot of places where `const` would be more applicable, but LGTM overall!

adeobas committed:
Update files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md

Co-authored-by: Queen Vinyl Da.i'gyu-Kazotetsu <vinyldarkscratch@gmail.com>

adeobas committed:
Update files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md

Co-authored-by: Queen Vinyl Da.i'gyu-Kazotetsu <vinyldarkscratch@gmail.com>

adeobas committed:
Update files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md

Co-authored-by: Queen Vinyl Da.i'gyu-Kazotetsu <vinyldarkscratch@gmail.com>

adeobas committed:
Update files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md

Co-authored-by: Queen Vinyl Da.i'gyu-Kazotetsu <vinyldarkscratch@gmail.com>

adeobas committed:
Update files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md

Co-authored-by: Queen Vinyl Da.i'gyu-Kazotetsu <vinyldarkscratch@gmail.com>

adeobas committed:
Update files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md

Co-authored-by: Queen Vinyl Da.i'gyu-Kazotetsu <vinyldarkscratch@gmail.com>

adeobas committed:
Update files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md

Co-authored-by: Queen Vinyl Da.i'gyu-Kazotetsu <vinyldarkscratch@gmail.com>

adeobas committed:
Update files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md

Co-authored-by: Queen Vinyl Da.i'gyu-Kazotetsu <vinyldarkscratch@gmail.com>

adeobas committed:
Update files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md

Co-authored-by: Queen Vinyl Da.i'gyu-Kazotetsu <vinyldarkscratch@gmail.com>

queengooborg committed:
Merge branch 'main' into patch-6

adeobas committed:
Reinsert Sidebar Call 

Updating page with sidebar

Page: Web Speech API 
Line: 14 
Call: {{DefaultAPISidebar("Web Speech API")}}

queengooborg reviewed (approved):
Thank you, LGTM!
